### DISCUSSION TITLE
Howto: New home screen

### DISCUSSION THREAD

**Discussion by ustraub** [jcorporation/myMPD#369] [Q&A] [answered]
When opening the mympd page a large grey panel appears with the text "You have not added any Home icons yet. You can add views, playlists, and scripts."
![grafik](https://user-images.githubusercontent.com/46170211/100279848-9271ca80-2f67-11eb-9b6a-8438fe715826.png)

**Comment by john-** [comment]
Looks like it is part of the new home page feature.   I see at least one place in other parts where one can add item to that page.  I think it was the search function.

**Comment by jcorporation** [comment]
Right, you can add home icons from all views, from the context menu of playlists and in the script window. I thought this message is self explanatory.

**Comment by ghost** [comment]
The new home screen is indeed a handy feature. I already created some views. Some not that good, which I want to delete. I can't seem to find a way to delete a unwanted view, other than removing them all by using:
`mympd reset_state`
Is there another way to remove views from the home screen?

**Comment by ustraub** [comment]
Hi,
it seemed obvious to me what I meant when opening this issue. Obviously it was unclear. So let me explain. I believe that this empty grey panel looks bad when you start the program the first time. And it is unclear what the purpose of this grey panel is and why it appears. 

From the comments I understand now that the grey panel is the empty home screen. But it is still unclear what the home screen is. From the comments it seems one can add some views. What are views? Links to albums? Links to songs? Links to a subpage? What happens when you click on a link? Can the home screen be removed/disabled? Is it always present? How do you add links? 

In general the possibility to customize the initial view is clearly an improvement. I am not criticizing its existence,

Sorry for my "dumb" questions, but I think that other newbies stumble over the same problems.

Regards
Ulrich Straub

**Comment by jcorporation** [comment]
@pinkdotnl: click on the text of an icon to open a context menu. From this you can edit or delete an icon. Reordering of the icons with drag and drop.

**Reply by onkobu** [reply]
For you as the maintainer the motivation and goal are clear. I'm not a myMPD newbie but I was struggling for at least 30min until I came here and had a closer look at the actions of various elements.

The feature itself is a great way to bookmark different things in one view. But from the initial sort of warning to the home screen editing is a great gap. Maybe the initial text should contain a sample/ illustration. I expected some coding to be necessary.

**Reply by jcorporation** [reply]
A link to a wiki page could be added.

**Reply by jcorporation** [reply]
![image](https://user-images.githubusercontent.com/765844/106062761-85a05d00-60f7-11eb-81ac-efe3ea428208.png)
Is this explanation helpful and comprehensible?

**Reply by Emilot** [reply]
It is ok, but i do not know about the "script" explanatory. I am not very sure that a lot of people know what "modal" is...

:)

**Reply by jcorporation** [reply]
Oh that could be true, I will revise the script explaination again

**Reply by onkobu** [reply]
Usability is not as easy as it seems but the sample shown gives the user options. A text telling me what I don't have doesn't get me out of the situation. With such ~breaking~innovative changes I appreciate those offering a way back to the classic layout – instead of suggesting instructions from the wiki.

I'd say there is a group in the target audience that has nothing to do with development and just wants to enjoy music. They maybe have a background/ expect a layout from common things like other players (hard- and software).

Another aspect is the display, e.g. mobile devices or a SoC with a tiny or medium sized touch screen vs. flat screens/ TVs or high res laptop monitors. (Imagine a media server hosting the application, one larger screen to do all the complex playlist editing and clients spread across the living space with only the playback visible/ necessary.)

**Reply by jcorporation** [reply]
I don’t know what you mean exactly. But you can disable the home screen in the configuration file.

The documentation is in the wiki. Why should a user not read this if he don’t know how a thing is working?

**Comment by jcorporation** [comment, marked as answer]
@ustraub:
The home screen is designed as an individual starting point. I added my favorite radio station (playlist), a script that enables the jukebox mode with my prefered playlist, a view to show the newest albums, ...

- the home screen can be disabled with a config option, the option is documented in the wiki.
- view: link to a view (subpage) with all options set as the view was added, e. g. browse database with album selected as tag and sorted by date. It behaves like a browser bookmark. Add this directly from the view with the add to home screen icon.
- playlist: the playlist replaces the current queue. Add this from the context menu of a playlist.
- Script: script to execute, add this from the script overview.

I hope this description helps. 

But a question: how should this home screen be designed to be more self explanatory?

I think no one would read a manual... or would dig through the issues here.

I suspect myMPD has meanwhile so many features and configurable options that it is not easy to understand. The advantage is the great flexibility. I try to preconfigure myMPD with useful default settings, but for getting the best out of myMPD reading the wiki or asking questions here is necessary.

**Comment by john-** [comment]
I was not expecting the home screen but felt the basic text "You have not added any Home Icons yet. You can add views, playlists and scripts." was enough for me to understand that if I wanted to do something I need to take an action.

There could be an argument made for some default entries that linked to the other screens (e.g. Playback).   These can be removed by those users that take the next step of customizing.   Yes, this would be redundant but would provide some context for those just starting out or those that have no desire to customize.
